libraries/recipe_helpers.rb

```diff
@@ -146,6 +150,8 @@ def service_provider(node)
             service_provider = Chef::Provider::Service.const_get(specified_provider)
           end
           service_provider
+        elsif systemd_platform?(node)
```

Comment:
We catch Amazon Linux 2023 first, to use Systemd, and then default to the previous test without modification

Reply:
Can this be added as a comment in code?